### PR TITLE
Fix getting polar axes in plt.polar()

### PR DESCRIPTION
**Pull request by dstansby**
Currently `ax = gca(polar=True)` raises a warning, because in future version it will return a new subplot instead of returning an existing one.

This PR removes this warning, and ensures `plt.polar` will behave well into the future by calling `gca()` without arguments. This means that if no axes are present, a new polar axes is manually created in `plt.polar`.

### CODE REVIEW

File: lib/matplotlib/figure.py [hunk 1]
```diff
@@ -1822,11 +1822,12 @@ def gca(self, **kwargs):
                         "3.0",
                         "Calling `gca()` using the same arguments as a "
                         "previous axes currently reuses the earlier "
-                        "instance.  In a future version, a new instance will "
-                        "always be created and returned.  Meanwhile, this "
-                        "warning can be suppressed, and the future behavior "
-                        "ensured, by passing a unique label to each axes "
-                        "instance.")
+                        "instance.\n"
```
Comment (dstansby) [inline lib/matplotlib/figure.py]: Also added some newlines here to make the warning more readable.

File: lib/matplotlib/pyplot.py [hunk 2]
```diff
@@ -2283,12 +2283,15 @@ def polar(*args, **kwargs):
...
-    # If an axis already exists, check if it has a polar projection
+    # If an axes already exists, check if it has a polar projection
     if gcf().get_axes():
         if not isinstance(gca(), PolarAxes):
             warnings.warn('Trying to create polar plot on an axis that does '
                           'not have a polar projection.')
-    ax = gca(polar=True)
+        ax = gca()
+    else:
+        ax = gcf().add_subplot(1, 1, 1, polar=True)
```
Comment (jklymak) [inline lib/matplotlib/pyplot.py]: umm, I'm not too up on the pyplot interface, but is `gca()` necessarilly a `subplot(111)`?

Comment (timhoffm) [reply lib/matplotlib/pyplot.py]: No, but: If there are axes, use the gca(). If not, make a new one equivalent to subplot(111).

Comment (jklymak) [reply lib/matplotlib/pyplot.py]: OK, but I don't get the point of this PR then.  Why is this better than whats there now?  Is the idea to let `plt.polar` write on an existing polar axes, despite the fact `gca` will create a new axes in the future?  That doesn't seem right.  Sorry if I'm being dense...

Comment (efiring) [reply lib/matplotlib/pyplot.py]: The whole `gca`situation is incredibly confusing.  I can't make sense of the current behavior and docstrings, and the warning about future behavior makes no sense to me either.  `gca` means "get current axes"; it should not mean "always get a new axes".  I think that the coming change to `subplot` is being incorrectly applied to `gca`.

Comment (dstansby) [reply lib/matplotlib/pyplot.py]: The change was made in https://github.com/matplotlib/matplotlib/pull/10660 - I think `gca` should just get the current axis, and not create a new one, and that change should be reverted. ping @anntzer

Comment (anntzer) [reply lib/matplotlib/pyplot.py]: I agree that the warning in #10660 is wrong.  I need to rethink about the correct way to ultimately get rid of the axes collision behavior.

Comment (dstansby) [reply lib/matplotlib/pyplot.py]: Great, I'll close this then.

**Pull request closed without merging**